xbmc/utils/Signals.h

```diff
@@ -0,0 +1,201 @@
+/*
```

Comment:
I think for now it is better to keep those things private to wayland. Kodi already has too many different approaches how to deal with signals/messages. 
IMO we should move to an actor model with asynchronous message passing.